python-sdk/pachyderm_sdk/api/cdr/resolver.py

```diff
@@ -118,16 +131,37 @@ def _dereference_compress(self, body: Compress) -> bytes:
         return gzip.decompress(inner)
 
     def _dereference_content_hash(self, body: ContentHash) -> bytes:
-        """Resolves a ContentHash CDR. This method must resolve its inner CDR."""
-        if body.algo != HashAlgo.BLAKE2b_256:
-            raise ValueError(f"unrecognized hash algorithm: {body.algo}")
-        inner = self.resolve(body.inner)
-        inner_hash = blake2b(inner, digest_size=32).digest()
-        if not compare_digest(inner_hash, body.hash):
-            raise ValueError(
-                f"content failed hash check. HAVE: {inner_hash} WANT: {body.hash}"
-            )
-        return inner
+        """Resolves a ContentHash CDR. This method must resolve its inner CDR.
+
+        If the cache_location has been set:
+          * the content will be read from disk if the content exists within the cache,
+            rather than resolving the HTTP reference.
+          * the content will be saved to the cache after resolving the HTTP reference.
+          * the cache uses the hex string of the content hash as its key (file name).
+        """
+
+        def _deref_inner() -> bytes:
+            if body.algo != HashAlgo.BLAKE2b_256:
```

Comment:
So do we only support BLAKE2b_256 as a hashing algorithm for ContentHash? I was kind of assuming since we have all this code around determining which algo was used we would be supporting multiple algos, but I could be thinking about this wrong.

Reply:
Yeah we currently only support one hashing algorithm, one compression algorithm, and one encryption algorithm. This is just coded defensively in case any new options get added before the python implementation can be updated.

Reply:
Makes sense. Defensively writing it to support multiple algos does not really add much overhead either so seems like a smart move.